galaxy/pulp/tasks.py

```diff
@@ -25,63 +25,57 @@
 
 from galaxy.main import models
 from galaxy.importer import collection as i_collection
+from galaxy.pulp import exceptions as exc
 from galaxy.importer import exceptions as i_exc
+from galaxy.common import logutils
 
 
 log = logging.getLogger(__name__)
 
 
-# TODO(cutwater): Move to exceptions module
-# TODO(cutwater): Better exception hierarchy
-class VersionConflict(Exception):
-    pass
+def import_collection(artifact_pk, repository_pk, namespace_pk, task_id):
+    log.info('Starting collection import task: {}'.format(task_id))
 
-
-class PulpTaskError(Exception):
-    pass
-
-
-def import_collection(artifact_pk, repository_pk, namespace_pk):
     artifact = pulp_models.Artifact.objects.get(pk=artifact_pk)
     repository = pulp_models.Repository.objects.get(pk=repository_pk)
+    import_task = models.ImportTask.objects.get(id=task_id)
+    log_db = _get_import_task_msg_logger(import_task)
+
+    import_task.start()
+    log_db.info('Starting import: task_id={}, artifact_pk={}'.format(
+                import_task.id, artifact_pk))
 
     try:
         with transaction.atomic():
-            collection_info = _process_collection(artifact)
-            _publish_collection(artifact, repository,
-                                namespace_pk, collection_info)
-    except Exception:
-        # Delete artifact file and model if import failed
-        artifact.file.delete(save=False)
-        artifact.delete()
+            collection_info = _process_collection(artifact, log_db)
+            coll, coll_ver = _publish_collection(artifact, repository,
+                                                 namespace_pk, collection_info)
+        _process_import_success(coll, coll_ver, import_task)
+    except i_exc.ImporterError as e:
+        _process_import_fail(artifact, import_task, msg=e)
+    except exc.VersionConflict:
+        msg = 'Collection version already exists in galaxy'
+        _process_import_fail(artifact, import_task, msg)
+    except Exception as e:
+        _process_import_fail(artifact, import_task, msg=e.__class__.__name__)
```

Comment:
It looks like this will drop a lot more information, like the `msg` of the original exception. Can that be preserved?

Reply:
On repository import an uncaught error says only “Task # failed”, here we give a bit more info (error name) but wanted to continue to avoid providing the galaxy stacktrace to the enduser.

Reply:
If the information gets clipped here to keep it from the user, can we at least log the exception data for ourselves?

Reply:
Ah yes, this wasn't getting into our pulp logs, thanks for the catch